petab/core.py

```diff
@@ -546,3 +550,48 @@ def get_notnull_columns(df, candidates):
     """
     return [col for col in candidates
             if col in df and not np.all(df[col].isnull())]
+
+
+def create_condition_df(parameter_ids, condition_ids=None):
+    """Create empty condition dataframe
+
+    Arguments:
+        parameter_ids: the columns
+        condition_ids: the rows
+    Returns:
+        An pandas.DataFrame with empty given rows and columns and all nan
+        values
+    """
+
+    data = {'conditionId': []}
+    for p in parameter_ids:
+        data[p] = []
+
+    df = pd.DataFrame(data)
+    df.set_index(['conditionId'], inplace=True)
+
+    if not condition_ids:
+        return df
+
+    for c in condition_ids:
+        df[c] = np.nan
```

Comment:
ah, that's what having the id as index is good for ...

Reply:
... and to make lookups faster.